src/components/SettingsModal.tsx

```diff
@@ -68,10 +106,19 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
 
   const onSave = () => {
     setSettings(localSettings);
+    const flowSettings = Object.entries(settings).filter(
+      ([key]) => key !== 'name' && key !== 'description',
+    );
+    if (localSettings.name != null) {
+      metadata.name = localSettings.name;
```

Comment:
This way of setting the value is working because we're mutating the object itself but given that is not changing the reference, the store doesn't recognize it, therefore, the source code sync is not working completely, as we can see when opening the source code editor after saving the configuration, there are pending changes to be synced (notice the banner at the top).

I think that it's better to leverage the store to conduct these changes. One way to do so would be to leverage [the existing `setMetadata`](https://github.com/KaotoIO/kaoto-ui/blob/7dfcbacd2366328f71e4d3bc970f6bffe4a8e44e/src/store/FlowsStore.ts#L198-L203) function from the flows store.

```typescript
setMetadata('name', localSettings.name);
setMetadata('description', localSettings.description)
```

Reply:
done :+1: